Tests/test_NCBIXML.py

```diff
@@ -1866,6 +1854,1681 @@ def test_xml_2218L_rpsblast_001(self):
+        self.assertEqual(alignments[3].title[:50], "gi|338714227|ref|XM_001492113.3| PREDICTED: Equus ")
+        self.assertEqual(alignments[3].length, 1390)
+        self.assertEqual(len(alignments[3].hsps), 2)
+        self.assertFalse(alignments[3].hsps[0].expect > E_VALUE_THRESH)
+        self.assertTrue(alignments[3].hsps[1].expect > E_VALUE_THRESH)
+        self.assertEqual(alignments[4].title[:50], "gi|390474391|ref|XM_002757683.2| PREDICTED: Callit")
+        self.assertEqual(alignments[4].length, 1402)
+        self.assertEqual(len(alignments[4].hsps), 2)
+        self.assertFalse(alignments[4].hsps[0].expect > E_VALUE_THRESH)
+        self.assertTrue(alignments[4].hsps[1].expect > E_VALUE_THRESH)
+
+        self.assertRaises(StopIteration, next, records)
+        handle.close()
+
+    def test_xml_2226_tblastn_006_err(self):
+        """Parse TBLASTN 2.2.26+."""
+        handle = StringIO("""<?xml version="1.0"?>
+                      <!DOCTYPE BlastOutput PUBLIC "-//NCBI//NCBI BlastOutput/EN" "http://www.ncbi.nlm.nih.gov/dtd/NCBI_BlastOutput.dtd">
+                        <BlastOutput>
+                          <BlastOutput_program>tblastn</BlastOutput_program>
+                          <BlastOutput_version>TBLASTN 2.2.26+</BlastOutput_version>
+                          <BlastOutput_reference>Stephen F. Altschul, Thomas L. Madden, Alejandro A. Sch&amp;auml;ffer, Jinghui Zhang, Zheng Zhang, Webb Miller, and David J. Lipman (1997), &quot;Gapped BLAST and PSI-BLAST: a new generation of protein database search programs&quot;, Nucleic Acids Res. 25:3389-3402.</BlastOutput_reference>
+                          <BlastOutput_db>db/minirefseq_mrna</BlastOutput_db>
+                          <BlastOutput_query-ID>Query_1</BlastOutput_query-ID>
+                          <BlastOutput_query-def>random_s00</BlastOutput_query-def>
+                          <BlastOutput_query-len>32</BlastOutput_query-len>
+                          <BlastOutput_param>
+                            <Parameters>
+                              <Parameters_matrix>BLOSUM62</Parameters_matrix>
+                              <Parameters_expect>10</Parameters_expect>
+                              <Parameters_gap-open>11</Parameters_gap-open>
+                              <Parameters_gap-extend>1</Parameters_gap-extend>
+                              <Parameters_filter>L;</Parameters_filter>
+                            </Parameters>
+                          </BlastOutput_param>
+                          <BlastOutput_iterations>
+                          </BlastOutput_iterations>
+                        </BlastOutput>
+                        """)
+
+        self.assertRaises(ValueError, NCBIXML.read, handle)
+        handle.close()
+
+    def test_xml_2226_tblastn_007_err(self):
+        """Parse TBLASTN 2.2.26+."""
+        handle = StringIO("""this is an empty file to test the method NCBIXML.handle()
```

Comment:
You've still got the confusing text here (about it being an empty file).

Also rather than embedding the string I was expecting you to do something like this:

```python
with open(some_good_blast_xml_filename) as handle:
   xml = handle.read()
handle = StringIO("some random text before the XML\n" + xml)
```